controllers/authController.js

```diff
@@ -8,41 +8,38 @@ const catchAsync = require("../util/async");
 const ApiError = require("../util/ApiError");
 
 const signup = catchAsync(async (req, res) => {
-    const session = await mongoose.startSession();
-    session.startTransaction();
-    try {
-        const user = await userService.createUser(req.body, { session });
-        const token = uuidv4();
-        await userService.updateVerificationToken(user, token, { session });
-        await emailService.sendVerificationEmail(user.email, token);
-        await session.commitTransaction();
-        return res.status(201).json({ success: true });
-    } catch (error) {
-        await session.abortTransaction();
-        throw new ApiError(error.statusCode, error.message, error.name);
-    } finally {
-        session.endSession();
-    }
+  const session = await mongoose.startSession();
+  session.startTransaction();
+  try {
+    const user = await userService.createUser(req.body, { session });
+    const token = uuidv4();
+    await userService.updateVerificationToken(user, token, { session });
+    await emailService.sendVerificationEmail(user.email, token);
+    await session.commitTransaction();
+    return res.status(201).json({ success: true });
+  } catch (error) {
+    await session.abortTransaction();
+    throw new ApiError(error.statusCode, error.message, error.name);
+  } finally {
+    session.endSession();
+  }
 });
 
 const login = catchAsync(async (req, res) => {
-    const { email, password } = req.body;
-    const user = await authService.loginUserWithEmailAndPassword(
-        email,
-        password
-    );
-    return res.status(200).json({ success: true, user: user.toJSON() });
+  const { email, password } = req.body;
+  const user = await authService.loginUserWithEmailAndPassword(email, password);
+  return res.status(200).json({ success: true, user: user.toJSON() });
 });
 
 const verifyEmail = catchAsync(async (req, res) => {
-    const user = await userService.verifyUser(req.params.token);
-    if (!user) {
-        res.status(204).json();
-    }
-    res.status(200).json({ success: true });
+  const user = await userService.verifyUser(req.params.token);
```

Comment:
use destructuring and also validation on token

Reply:
Could you please provide what kind of validation should be applied on token

Reply:
destructure it and check if it is coming or not from the frontend

Reply:
I have passed the request object through joi validation. It checked fro token